File: qualia-tempo-prototype/frontend/src/services/contracts/ILogger.contracts.ts
/**
 * QUALIA.CODE v1.1 - Logger Contracts
 * Centralized type definitions for logging system
 *
 * Purpose: Single source of truth for all logging-related data structures
 * Architecture: Contract definitions extracted from service implementation for clarity and reusability
 */

// Log entry interface for structured logging
export interface LogEntry {
  level: LogLevel;
  message: string;
  context?: Record<string, unknown>;
  timestamp: string;
  source: string;
}

// Log levels for filtering and prioritization
// eslint-disable-next-line no-unused-vars
export enum LogLevel {
  // eslint-disable-next-line no-unused-vars
  DEBUG = 0,
  // eslint-disable-next-line no-unused-vars
  INFO = 1,
  // eslint-disable-next-line no-unused-vars
  WARN = 2,
  // eslint-disable-next-line no-unused-vars
  ERROR = 3,
  // eslint-disable-next-line no-unused-vars
  NONE = 4,
}

// Logger Configuration - Migrated from ConfigurationService.ts
export interface LoggerConfig {
  level: LogLevel;
  enableConsole: boolean;
  enableFile: boolean;
  enableRemote: boolean;
  maxFileSize: number;
  maxFiles: number;
  logDirectory: string;
  remoteEndpoint?: string;
  format: 'json' | 'text';
  timestampFormat: string;
}

File: qualia-tempo-prototype/backend/services/contracts/ILogger_contracts.py
# QUALIA.CODE v1.1 - ILogger Contracts
from dataclasses import dataclass

@dataclass
class LoggerConfig:
    """Configuration contract for Logger."""
    log_level: str = "INFO"
    log_format: str = "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
    enable_file_logging: bool = True
    log_file_path: str = "backend.log"
    max_file_size_mb: int = 10
    backup_count: int = 5
